### ISSUE TITLE
Error 3 on iOS from showUrl when trying to do login in some conditions

### ISSUE THREAD

**Issue by ghills** [auth0/react-native-auth0#382]
### Describe the problem

Authorize call doesn't prompt for login when it should on iOS hardware. With changes in #381 I see the following error occuring from the iOS native code:

`LOG  {"error": "The operation couldn’t be completed. (com.apple.AuthenticationServices.WebAuthenticationSession error 3.)", "error_code": 3}`

It doesn't seem to happen on initial login. It is happening when I am trying to migrate our app to a new version where we added a couple of new scopes to the request. I have been able to reproduce by deleting the app from my phone and starting over (which results in a successful login) that only breaks when I attempt to add scopes.

### What was the expected behavior?

Prompted for login via auth0

### Reproduction

Here is the call to authorize:

```javascript
.authorize({
        audience: 'https://foo.bar.com/api',
        scope: 'offline_access openid email profile',
    }, 
    { ephemeralSession: true })
```

1. Log in with a given set of scopes.
2. Close the running app.
3. Change the code to add a new scope and rebuild.
4. Run again and attempt to login

### Environment

- **Version of this library used:** 2.8.3
- **Which framework are you using, if applicable:** React Native 0.64.1, iOS 14.4.2
- **Other modules/plugins/libraries that might be involved:** 
- **Any other relevant information you think would be useful:** Unable to reproduce on the iOS simulator, only on device.

**Comment by Widcket** [comment]
Hi @ghills, thanks for raising this. Do you have any rules/hooks/actions on your tenant that might be causing auth to fail?

**Comment by ghills** [comment]
> Hi @ghills, thanks for raising this. Do you have any rules/hooks/actions on your tenant that might be causing auth to fail?

Hello @Widcket - thank you for your response. We do have a few rules that run but I don't think they are failing and I don't see any indication in the logs or anything related to the attempted login at all. To be clear, when this happens I don't get any popup to even attempt to log in at all.

**Comment by Widcket** [comment]
That seems like an `ASWebAuthenticationSession` error, particularly `The UIWindowScene for the returned window was not in the foreground active state.` (error code 3).
By any chance, is this an app that hasn't been updated (or tested on an iOS device) for a long time? Because that error is related to some iOS 13 changes.
And,per chance, are you invoking WebAuth on `viewDidLoad`?

**Comment by ghills** [comment]
> That seems like an `ASWebAuthenticationSession` error, particularly `The UIWindowScene for the returned window was not in the foreground active state.` (error code 3).
> By any chance, is this an app that hasn't been updated (or tested on an iOS device) for a long time? Because that error is related to some iOS 13 changes.
> And,per chance, are you invoking WebAuth on `viewDidLoad`?

It's not an overly outdated app - I have made recent changes to it over the last year that have been tested on device.

However, your second question I think may be on to something with it being timing related. This is happening from react native so it isn't in `viewDidLoad` itself but I think in this case we were having UI changes happen at the same time as the login was initiated (Auth0 was actually being called from an async process outside of the component lifecycle).

To confirm this I tried adding an intentional delay before calling `authorize` and that seems to alleviate the error which leads me to believe it has to do with what views were on the screen when the login was initiated changing.

Assuming my understanding is correct I will head down that path of restructuring that workflow to make sure the views are not changing at the time of the call to `authorize`.

Thanks for your help!
